cpu/stm32f4/include/periph_cpu.h

```diff
@@ -136,6 +150,21 @@ typedef struct {
     uint8_t dev;            /**< ADCx - 1 device used for the channel */
     uint8_t chan;           /**< CPU ADC channel connected to the pin */
 } adc_conf_t;
+/** @} */
+
+/**
+ * @brief   I2C configuration options
+ */
+typedef struct {
+    I2C_TypeDef *dev;       /**< I2C device */
+    gpio_t scl_pin;         /**< clock pin */
+    gpio_t sda_pin;         /**< data pin */
+    uint8_t scl_af;         /**< SCL pin alternate function */
+    uint8_t sda_af;         /**< SDA pin alternate function */
+    i2c_speed_t speed;      /**< I2C bus speed */
+    uint8_t rcc_bit;        /**< bit in the rcc register */
```

Comment:
as said, those are just hacked together to demonstrate the API... Will be touched anyway